Amplify/Core/Support/Internal/NSLocking+Execute.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// - Warning: Although this has `public` access, it is intended for internal use
+/// and should not be used directly by host applications. The behaviors and names of
+/// this type may change without warning.
+extension NSLocking {
```

Comment:
also im stealing this :)

Reply:
It's open-source. All we need is attribution. :P